src/bruteforce/bruteforce-omp_ma.c

```diff
@@ -0,0 +1,74 @@
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <omp.h>
+#include "../hash/hash.h"
+#include "bruteforce-util.c"
+
+int bruteforce_crack(char *password_hash, char *characters, int password_max_length, int verbose)
+{
+    int number_of_characters = strlen(characters);
+    printf("Brute force of hash: %s\n", password_hash);
+    printf("Using %d characters: %s\n", number_of_characters, characters);
+    printf("Calculating to a length of %d\n", password_max_length);
+
+    int result = 1;
+
+    static unsigned char buffer[65];
+    int i, j, k;
+
+    for (i = 1; i <= password_max_length; i++)
+    {
+        long possibilities = (long)pow(number_of_characters, i);
+        if (verbose)
+        {
+            printf("Now calculating password length of %d, it has %ld possibilities\n", i, possibilities);
+        }
+
+#pragma omp parallel private(buffer, j, k) shared(result)
+        {
+            char passwordToTest[i + 1];
+            for (j = 0; j < possibilities;)
+            {
+                if (result == 0)
+                {
+                    printf("Found pass already...\n");
+                    //return; //Skip the next processes if we have found the result
+                }
+                int nextStep = j + 100000;
+                int p = omp_get_thread_num();
+                printf("Rank: %d processing",p);
+                #pragma omp for schedule(auto)
+
+                for (j = j; j < nextStep; j += p)
```

Comment:
Change the j += p to just a j++, we let the omp compiler handle breaking up the for loop into chunks.
I also think the j=j is not necessary.